spec/models/framework_response_spec.rb

```diff
@@ -227,5 +227,163 @@
 
       expect(response.flags).to contain_exactly(flag2)
     end
+
+    context 'when dependent responses' do
+      it 'clears dependent responses if value changed' do
+        parent_response = create(:string_response)
+        child1_question = create(:framework_question, dependent_value: 'Yes', parent: parent_response.framework_question)
+        child2_question = create(:framework_question, dependent_value: 'Yes', parent: parent_response.framework_question)
+        create(:string_response, framework_question: child1_question, parent: parent_response)
+        create(:string_response, framework_question: child2_question, parent: parent_response)
+        parent_response.update_with_flags!('No')
+
+        parent_response.dependents.each do |child_response|
```

Comment:
maybe add a test to very parent_response has dependents, as this test will skipped if it does not